src/main/java/sirius/biz/tycho/dashboards/BaseTimeseriesChart.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.tycho.dashboards;
+
+import sirius.biz.analytics.charts.ComparisonPeriod;
+import sirius.biz.analytics.charts.Dataset;
+import sirius.biz.analytics.metrics.MetricQuery;
+import sirius.web.services.JSONStructuredOutput;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Provides basic functionality needed by all time-series based charts.
+ */
+public abstract class BaseTimeseriesChart implements Chart {
+
+    private List<String> labels = Collections.emptyList();
```

Comment:
Shouldn't this be mutable and final and be modified/filled in the methods below?

Reply:
Yes but the current implementation allows to overwrite the used implementation of the labels List, so someone could pass an immutable list or another one a mutable ArrayList.

Wouldn't it follow best practices when the actual List implementation used was defined here and only the entries of the passed in list were copied in `withLabels` via `addAll`?